testkit/tests/supervisor/main.rs

```diff
@@ -0,0 +1,795 @@
+    assert!(!does_artifact_exist(&api, &artifact.name));
+}
+
+#[test]
+fn test_empty_service_instance_name() {
+    let mut testkit = testkit_with_inc_service();
+    deploy_default(&mut testkit);
+
+    let api = testkit.api();
+    let artifact = artifact_default();
+    let empty_instance_name = "";
+    let deadline_height = testkit.height().next();
+    let request = start_service_request(artifact, empty_instance_name, deadline_height);
+    let hash = start_service(&api, request);
+    testkit.create_block();
+
+    let system_api = api.exonum_api();
+    let expected_status = json!({
+       "type": "service_error",
+       "code": 7,
+       "description": "Service instance name should not be empty",
+    });
+    system_api.assert_tx_status(hash, &expected_status);
+}
+
+#[test]
+fn test_bad_service_instance_name() {
+    let mut testkit = testkit_with_inc_service();
+    deploy_default(&mut testkit);
+
+    let api = testkit.api();
+    let artifact = artifact_default();
+    let bad_instance_name = "\u{2764}";
+    let deadline_height = testkit.height().next();
+    let request = start_service_request(artifact, bad_instance_name, deadline_height);
+    let hash = start_service(&api, request);
+    testkit.create_block();
+
+    let system_api = api.exonum_api();
+    let expected_status = json!({
+       "type": "service_error",
+       "code": 7,
+       "description": "Service instance name contains illegal character, use only: a-zA-Z0-9 and one of _-.",
+    });
+    system_api.assert_tx_status(hash, &expected_status);
+}
+
+#[test]
+fn test_start_service_instance_twice() {
+    let instance_name = "inc";
+    let mut testkit = testkit_with_inc_service();
+    deploy_default(&mut testkit);
+
+    // Start the first instance
+    {
+        let api = testkit.api();
+        assert!(!does_service_instance_exist(&api, instance_name));
+
+        let deadline = testkit.height().next();
+        let request = start_service_request(artifact_default(), instance_name, deadline);
+        let hash = start_service(&api, request);
+        testkit.create_block();
+
+        api.exonum_api().assert_tx_success(hash);
+
+        let api = testkit.api(); // Update the API
+        assert!(does_service_instance_exist(&api, instance_name));
+    }
+
+    // Try to start another instance with the same name
+    {
+        let api = testkit.api();
+
+        let deadline = testkit.height().next();
+        let request = start_service_request(artifact_default(), instance_name, deadline);
+        let hash = start_service(&api, request);
+        testkit.create_block();
+
+        let system_api = api.exonum_api();
+        let expected_status = json!({
+           "type": "service_error",
+           "code": 3,
+           "description": "Instance with the given name already exists.",
+        });
+        system_api.assert_tx_status(hash, &expected_status);
+    }
+}
+
+#[test]
+fn test_restart_node_and_start_service_instance() {
```

Comment:
Add also test for restart during the artifact deployments.

Reply:
Added a draft test